addons/binding/org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/cluster/ZigBeeClusterHandler.java

```diff
@@ -0,0 +1,152 @@
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.zigbee.ZigBeeBindingConstants;
+import org.openhab.binding.zigbee.handler.ZigBeeCoordinatorHandler;
+import org.openhab.binding.zigbee.handler.ZigBeeThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * ZigBeeClusterConverter class. Base class for all converters that convert between ZigBee clusters and openHAB
+ * channels.
+ *
+ * @author Chris Jackson
+ */
+public abstract class ZigBeeClusterHandler {
+    private static Logger logger = LoggerFactory.getLogger(ZigBeeClusterHandler.class);
+
+    protected ZigBeeThingHandler thing = null;
+    // protected ZigBeeThingChannel channel = null;
+    protected ZigBeeCoordinatorHandler coordinator = null;
+
+    protected ChannelUID channelUID = null;
+    protected String address = null;
+
+    private static Map<Integer, Class<? extends ZigBeeClusterHandler>> clusterMap = null;
+
+    /**
+     * Constructor. Creates a new instance of the {@link ZWaveCommandClassConverter} class.
+     *
+     */
+    public ZigBeeClusterHandler() {
+        super();
+    }
+
+    public void createConverter(ZigBeeThingHandler thing, ChannelUID channelUID, ZigBeeCoordinatorHandler coordinator,
+            String address) {
+        this.thing = thing;
+        this.channelUID = channelUID;
+        this.address = address;
+        this.coordinator = coordinator;
+    }
+
+    public abstract void initializeConverter();
+
+    public void disposeConverter() {
+    }
+
+    /**
+     * Execute refresh method. This method is called every time a binding item is refreshed and the corresponding node
+     * should be sent a message.
+     *
+     * @param channel the {@link ZigBeeThingChannel}
+     */
+    public void handleRefresh() {
+    }
+
+    /**
+     * Receives a command from openHAB and translates it to an operation on the Z-Wave network.
+     *
+     * @param channel the {@link ZigBeeThingChannel}
+     * @param command the {@link Command} to send
+     */
+    public void handleCommand(Command command) {
+    }
+
+    public abstract List<Channel> getChannels(ThingUID thingUID, Device device);
+
+    /**
+     *
+     * @param clusterId
+     * @return
+     */
+    public static ZigBeeClusterHandler getConverter(int clusterId) {
+        if (clusterMap == null) {
+            clusterMap = new HashMap<Integer, Class<? extends ZigBeeClusterHandler>>();
+
+            // Add all the handlers into the map...
+            clusterMap.put(ZigBeeApiConstants.CLUSTER_ID_ON_OFF, ZigBeeOnOffClusterHandler.class);
+            clusterMap.put(ZigBeeApiConstants.CLUSTER_ID_LEVEL_CONTROL, ZigBeeLevelClusterHandler.class);
+            clusterMap.put(ZigBeeApiConstants.CLUSTER_ID_COLOR_CONTROL, ZigBeeColorClusterHandler.class);
+            clusterMap.put(ZigBeeApiConstants.CLUSTER_ID_RELATIVE_HUMIDITY_MEASUREMENT,
+                    ZigBeeRelativeHumidityMeasurementClusterHandler.class);
```

Comment:
@cdjackson Seems that ZigBeeRelativeHumidityMeasurementClusterHandler.class
 & ZigBeeTemperatureMeasurementClusterHandler.class are missing in your PR